builder/confd/confd.go

```diff
@@ -0,0 +1,95 @@
+// Package confd provides basic Confd support.
+//
+// Right now, this library is highly specific to the needs of the present
+// builder. Because the confd library is not all public, we don't use it directly.
+// Instead, we invoke the CLI.
+package confd
+
+import (
+	"github.com/Masterminds/cookoo"
+	"github.com/Masterminds/cookoo/log"
+	"github.com/Masterminds/cookoo/safely"
+
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+)
+
+// defaultEtcd is the default Etcd host.
+const defaultEtcd = "127.0.0.1:4001"
+
+// RunOnce runs the equivalent of `confd --onetime`.
+//
+// This may run the process repeatedly until either we time out (~20 minutes) or
+// the templates are successfully built.
+//
+// Importantly, this blocks until the run is complete.
+//
+// Params:
+// - node (string): The etcd node to use. (Only etcd is currently supported)
+//
+// Returns:
+// - The []bytes from stdout and stderr when running the program.
+//
+func RunOnce(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+
+	node := p.Get("node", defaultEtcd).(string)
+
+	dargs := []string{"-onetime", "-node", node, "-log-level", "error"}
+
+	log.Info(c, "Building confd templates. This may take a moment.")
+
+	limit := 1200
+	timeout := time.Second * 3
+	var lasterr error
+	start := time.Now()
+	for i := 0; i < limit; i++ {
+		if out, err := exec.Command("confd", dargs...).CombinedOutput(); err == nil {
+			log.Infof(c, "Templates generated for %s on run %d", node, i)
+			return out, nil
+		} else {
```

Comment:
I intentionally ignored lint on that one. If you look at the code, you'll see why. `out` is not in scope without the else. IIRC, `out` is opened for GC as long as it's in the else block because the sleep call will yield to the scheduler.

I don not tend to blindly follow golint's suggestions, though I do consider them.